packages/access-api/src/routes/validate-email.js

```diff
@@ -134,6 +138,38 @@ async function session(req, env) {
     req.query.ucan,
     delegation.capabilities[0].nb.key
   )
+  if (req.method.toLowerCase() === 'post') {
+    const accessSessionResult = await validator.access(delegation, {
+      capability: Access.session,
+      principal: Verifier,
+      authority: env.signer,
+    })
+    if (accessSessionResult.error) {
+      throw new Error(
+        `unable to validate access session: ${accessSessionResult.error}`
+      )
+    }
+    // generate a delegation to the key that we can save in
+    // models.delegations to be found by subsequent access/claim
+    // invocations invoked by the did:key
+    const delegateToKey = await ucanto.delegate({
+      issuer: env.signer,
+      audience: {
```

Comment:
Audience of `./update` capabilities is meant to be non `did:key` principal, because those are only times ucanto would look at `./update` capabilities. Delegation to the agent key is not going to get you anything other than been able to claim that delegation from that agent afterwards.